Xamarin.PropertyEditing.Mac/Controls/Custom/UnderlinedTabViewController.cs

```diff
@@ -182,27 +182,34 @@ private void SetSelected (int index)
 				this.selected.Selected = true;
 		}
 
-		private NSView GetView (NSTabViewItem item)
+		private NSView GetView (NSTabViewItem item, nint index)
 		{
 			var id = item.Identifier as NSObjectFacade;
-			NSView tabView;
+			TabButton tabButton;
 			if (id != null) {
-				tabView = new UnderlinedImageView (HostResources, ((string)id.Target)) {
-					Selected = this.tabStack.Views.Length == SelectedTabViewItemIndex,
-					ToolTip = item.ToolTip
-				};
-			} else {
-				tabView = new UnderlinedTextField {
-					BackgroundColor = NSColor.Clear,
-					Editable = false,
-					Bezeled = false,
-					StringValue = item.Label,
-					Selected = this.tabStack.Views.Length == SelectedTabViewItemIndex,
-					ToolTip = item.ToolTip
-				};
+				tabButton = new TabButton (HostResources, (string)id.Target);
+			} else {
+				tabButton = new TabButton (HostResources) {
+					ControlSize = NSControlSize.Mini,
+					Font = NSFont.SystemFontOfSize (NSFont.SystemFontSizeForControlSize (NSControlSize.Mini)),
+					Title = item.Label,
+				};
 			}
 
-			return tabView;
+			tabButton.Tag = index;
+			tabButton.Selected = this.tabStack.Views.Length == SelectedTabViewItemIndex;
+			tabButton.ToolTip = item.ToolTip;
+			tabButton.Clicked += OnTabButtonClicked;

```

Comment:
What unregisters this event handler?

Reply:
It will be unregistered in the RemoveTab override, before the view is disposed.